api/v1alpha3/machinehealthcheck_webhook.go

```diff
@@ -36,8 +37,21 @@ var (
 	// 10 minutes should allow the instance to start and the node to join the
 	// cluster on most providers.
 	defaultNodeStartupTimeout = metav1.Duration{Duration: 10 * time.Minute}
+	// Minimum time allowed for a node to start up
+	minNodeStartupTimeout     = metav1.Duration{Duration: 30 * time.Second}
+	minNodeStartupTimeoutOnce sync.Once
 )
 
+// SetMinNodeStartupTimeout allows users to optionally set a custom timeout
+// for the validation webhook.
+// This function is mostly used within envtest (integration tests), and should
+// never be used in a production environment.
+func SetMinNodeStartupTimeout(d metav1.Duration) {
+	minNodeStartupTimeoutOnce.Do(func() {
+		minNodeStartupTimeout = d
+	})
```

Comment:
@vincepri, this code doesn't work unless `GODEBUG=asyncpreemptoff=1` is used.

Reply:
Is this related to the CI failures? We can avoid using sync.Once, and do a plain unsafe set. We can fix it later using the `atomic` package

Reply:
Yeah, it's causing the CI failures. Have no idea why it's making the tests fail, it's just if I remove it it works again.